lib/semantics/resolve-names.cc

```diff
@@ -2790,6 +2791,22 @@ Symbol &SubprogramVisitor::PushSubprogramScope(
   return *symbol;
 }
 
+Symbol &SubprogramVisitor::PushBlockDataScope(const parser::Name &name) {
+  if (auto *prev{FindSymbol(name)}) {
+    if (prev->attrs().test(Attr::EXTERNAL) && prev->has<ProcEntityDetails>()) {
+      if (prev->test(Symbol::Flag::Subroutine) ||
+          prev->test(Symbol::Flag::Function)) {
+        Say2(name, "BLOCK DATA '%s' has been called"_err_en_US, *prev,
+            "Previous call of '%s'"_en_US);
+      }
+      EraseSymbol(name);
```

Comment:
Are you erasing the symbol because EXTERNAL cannot be in BlockData? If yes, are we printing the error message?

Reply:
Ok, I am wrong, this is the implementation of:
There shall not be more than one unnamed block data program unit in a program.